Undeclared variable

After https://github.com/ros2/ros2/pull/793 I tried re-building my whole workspace, and it failed at `fastrtps`:

```
/home/user/ros2_ws/src/eProsima/Fast-RTPS/src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp: In member function ‘bool eprosima::fastrtps::rtps::EDP::pairingWriter(eprosima::fastrtps::rtps::RTPSWriter*, const eprosima::fastrtps::rtps::GUID_t&, const eprosima::fastrtps::rtps::WriterProxyData&)’:
/home/user/ros2_ws/src/eProsima/Fast-RTPS/src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp:967:80: error: ‘writer_guid’ was not declared in this scope
                                 update_publication_matched_status(reader_guid, writer_guid, 1);
                                                                                ^~~~~~~~~~~
/home/user/ros2_ws/src/eProsima/Fast-RTPS/src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp:967:80: note: suggested alternative: ‘reader_guid’
                                 update_publication_matched_status(reader_guid, writer_guid, 1);
                                                                                ^~~~~~~~~~~
                                                                                reader_guid
```

Seems like this was introduced by https://github.com/eProsima/Fast-RTPS/pull/613.

--- a/src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp
+++ b/src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp
@@ -962,7 +962,7 @@ bool EDP::pairingWriter(
                         info.remoteEndpointGuid = reader_guid;
                         W->getListener()->onWriterMatched(W,info);
 
-
+                        const GUID_t& writer_guid = W->getGuid();
                         const PublicationMatchedStatus& pub_info =
                                 update_publication_matched_status(reader_guid, writer_guid, 1);
                         W->getListener()->onWriterMatched(W, pub_info);

